infer/environment.cc

```diff
@@ -1061,13 +1061,12 @@ core::TypePtr Environment::processBinding(core::Context ctx, cfg::Binding &bind,
                                                                  ty.origins2Explanations(ctx)));
                         }
                     }
-                } else {
+                } else if (!c->isSynthetic) {
```

Comment:
the change makes sense, but I'm curios how did your prior commit cause this bug. Do you understand it?

Reply:
Yup! Now that we have a `static-init` there we end up with it having this in the CFG:
```
<self>: T.untyped = cast(<self>: NilClass, T.untyped);
```
made by this code https://github.com/sorbet/sorbet/blob/pt-stop-eliding/cfg/builder/builder_entry.cc#L33-L36 and we aren't happy if a user writes that.

Reply:
(only when doing this file because the class name becomes `T.untyped` gems/sorbet-runtime/lib/types/runtime_profiled.rb)

Reply:
Hm, what class has T.untyped as `selfClaz.data(ctx)->enclosingClass(ctx).data(ctx)->selfType(ctx)`?

Untyped itself?